src/components/ReportActionItem/MoneyRequestView.js

```diff
@@ -228,6 +229,43 @@ function MoneyRequestView({report, parentReport, parentReportActions, policyCate
     const pendingAction = lodashGet(transaction, 'pendingAction');
     const getPendingFieldAction = (fieldPath) => lodashGet(transaction, fieldPath) || pendingAction;
 
+    const getErrorForField = useCallback(
+        (field) => {
+            // Checks applied when creating a new money request
+            // NOTE: receipt field can return multiple violations, so we need to handle it separately
+            const fieldChecks = {
+                amount: {
+                    isError: transactionAmount === 0,
+                    translationPath: 'common.error.enterAmount',
+                },
+                merchant: {
+                    isError: isPolicyExpenseChat && isEmptyMerchant,
+                    translationPath: 'common.error.enterMerchant',
+                },
+                date: {
+                    isError: transactionDate === '',
+                    translationPath: 'common.error.enterDate',
+                },
+            };
+
+            const {isError, translationPath} = fieldChecks[field] ? fieldChecks[field] : {};
+
+            // Display form errors when first creating the money request
+            if (hasErrors && isError && translationPath) {
+                return translate(translationPath);
+            }
+
+            // Show violations if there are any
+            if (canUseViolations && hasViolations(field)) {
+                const violations = getViolationsForField(field);
+                return ViolationsUtils.getViolationTranslation(violations[0], translate);
+            }
+
+            return '';
```

Comment:
Do we want to add new code treating `''` as "not set"?

Reply:
I've kept it how it was before, which I think is the sensible choice given that we're in javascript and not typescript, so we don't hav guarantees about how other code calls the component. 

The empty string is falsy, so it'll behave the same way as `null` or `undefined` in any boolean checks, and this will preserve that behaviour.

Previously the empty string was used to mean "not set" as in this example
```ts
error={hasErrors && transactionAmount === 0 ? translate('common.error.enterAmount') : ''}
```


And when it's rendered in `<MenuItem>`, the double shebang `!!('')` essentially casts the value to a boolean so it will gracefully handle booleans, strings, `null` and `undefined`.
```tsx
  {!!errorText && (
   <FormHelpMessage
      isError
      shouldShowRedDotIndicator={false}
      message={errorText}      
      style={styles.menuItemError}
    />
)}
```